File: Python/1437_kLengthApart.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        lastOneIndex = -1
        for i in range(len(nums)):
            if nums[i] == 1:
                if lastOneIndex != -1 and i - lastOneIndex -1 < k:
                    return False
                lastOneIndex = i

        return True

File: C++/1437_kLengthApart.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        char lastOneIndex = -1;
        for(char i = 0; i < nums.size(); i++){
            if(nums[i] == 1){
                if(lastOneIndex != -1 && i - lastOneIndex -1 < k){
                    return false;
                }
                lastOneIndex = i;
            }
        }
        return true;
    }
};
